compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributedInterfaceSupertypeGenerator.kt

```diff
@@ -244,17 +240,10 @@ internal class ContributedInterfaceSupertypeGenerator(
    * We're not able to get 1:1 parity due to some type restrictions but it should be enough to make
    * the migration much more feasible in large projects that have a lot of ranked bindings.
    *
-   * There are two important limitations to note here:
-   * 1. Ranked bindings and bindings that get outranked must both explicitly declare their binding
-   *    type in order for us to actually compare them, because supertypes are not resolvable here.
-   *    The user will end up getting a duplicate binding error if the outranked binding is using an
-   *    implicit type.
-   * 2. We can't check for qualifiers when comparing these bindings because those annotation calls
-   *    are not resolved at this point. E.g. a compiler critical annotation
-   *    like @ContributesBinding(..) will be resolved but @Named(..) will not. This means that
-   *    qualifiers are unsupported for rank interop support. Rank can only effectively be used for
-   *    binding types where all of them are unqualified or all use the same qualifier. Other
-   *    combinations will need to be migrated to instead use explicit replacements or exclusions.
+   * The main limitation we have is that supertypes are not resolvable here. That means ranked
+   * bindings and bindings that get outranked must both explicitly declare their binding type for us
+   * to actually compare them. The user will end up getting a duplicate binding error if the
+   * outranked binding is using an implicit type.
```

Comment:
Can you unpack this one a bit more for me here? Or maybe add an example snippet to the kdoc here?

Reply:
Took a stab at making this clearer, lmk if it's helpful or could use more clarification

Reply:
lgtm!